pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -366,40 +368,30 @@ func (r *ReconcileHyperConverged) getHyperConverged(req *common.HcoRequest) (*hc
 }
 
 // updateHyperConverged updates the HyperConverged resource according to its state in the request.
-func (r *ReconcileHyperConverged) updateHyperConverged(request *common.HcoRequest) error {
+func (r *ReconcileHyperConverged) updateHyperConverged(request *common.HcoRequest) (bool, error) {
 
 	// Since the status subresource is enabled for the HyperConverged kind,
 	// we need to update the status and the metadata separately.
 	// Moreover, we need to update the status first, in order to prevent a conflict.
 	// In addition, spec changes are removed by status update, but since status update done first, we need
 	// to store the spec and recover it after status update
 
-	var spec hcov1beta1.HyperConvergedSpec
 	if request.Dirty {
-		request.Instance.Spec.DeepCopyInto(&spec)
+		err := r.updateHyperConvergedSpecMetadata(request)
+		if err != nil {
+			r.logHyperConvergedUpdateError(request, err, "Failed to update HCO CR")
```

Comment:
maybe be more specific:
"Failed to update HCO CR spec and metadata"

Reply:
We're logging the original error. I think it's too much information for the user.